src/snowflake/snowpark/session.py

```diff
@@ -1470,18 +1469,14 @@ def create_dataframe(
         if isinstance(data, Row):
             raise TypeError("create_dataframe() function does not accept a Row object.")
 
-        if not isinstance(data, (list, tuple, pandas.DataFrame)):
+        if not isinstance(data, (list, tuple)) and (
+            not installed_pandas
+            or (installed_pandas and not isinstance(data, pandas.DataFrame))
+        ):
             raise TypeError(
                 "create_dataframe() function only accepts data as a list, tuple or a pandas DataFrame."
             )
 
-        if not installed_pandas and isinstance(data, pandas.DataFrame):
-            raise MissingDependencyError(
-                "snowflake-connector-python[pandas]. create_dataframe() function only accepts data as a pandas "
-                "DataFrame when the Snowflake Connector for Python is the Pandas-compatible version. Please install "
-                'it as follow: `pip install "snowflake-connector-python[pandas]"`'
-            )
```

Comment:
in the original PR, was there a case where we were passing a pandas dataframe without installing pandas? I would think is it not possible to send a pandas dataframe if pandas is not installed